### PR TITLE
Fix/exclude indicator years

### PR DESCRIPTION
**Pull request by mariamiah**

### CODE REVIEW

File: utils/select.ts [hunk 1]
```diff
@@ -29,24 +29,33 @@ const createIndicatorOptionsFromTheme = (theme: SpotlightTheme): SelectOptions =
...
-export const createYearOptionsFromRange = (startYear = 0, endYear = 0, _exclude?: number[]): SelectOptions => {
+export const createYearOptionsFromRange = (startYear = 0, endYear = 0, excludedYear: number[]): SelectOptions => {
   const options: SelectOption[] = [];
   if (startYear || endYear) {
     const difference = Math.abs(endYear - startYear);
     for (let i = 0; i <= difference; i++) {
       const year = startYear + i;
-      options.push({ value: `${year}`, label: `${year}` });
+      if (excludedYear) {
+        if (!excludedYear.includes(year)) {
```
Comment (edwinmp) [inline utils/select.ts:39]: Rewrite with a single `if` statement.

File: utils/select.ts [hunk 2]
```diff
@@ -29,24 +29,33 @@ const createIndicatorOptionsFromTheme = (theme: SpotlightTheme): SelectOptions =
...
 
-export const createYearOptionsFromIndicator = ({ start_year, end_year }: SpotlightIndicator): SelectOptions => {
+export const createYearOptionsFromIndicator = ({
+  start_year,
+  end_year,
+  excluded_years
+}: SpotlightIndicator): SelectOptions => {
   const startYear = start_year || end_year || 0;
   const endYear = end_year || start_year || 0;
+  const excludedYear = excluded_years || [];
```
Comment (edwinmp) [inline utils/select.ts:56]: You won't need this once u change it to a string.

File: components/RevenueExpenditureSection/RevenueExpenditureSection.tsx [hunk 3]
```diff
@@ -127,7 +127,11 @@ const RevenueExpenditureSection: FunctionComponent<RevenueSectionProps> = ({ ind
           <FormField className="form-field--inline">
             <FormFieldSelect
               label="Year"
-              options={createYearOptionsFromRange(indicator.start_year, indicator.end_year)}
+              options={createYearOptionsFromRange(
+                indicator.start_year,
+                indicator.end_year,
+                indicator.excluded_years as string
```
Comment (edwinmp) [inline components/RevenueExpenditureSection/RevenueExpenditureSection.tsx:133]: You don't need to cast ...

File: utils/select.ts [hunk 4]
```diff
@@ -29,24 +29,30 @@ const createIndicatorOptionsFromTheme = (theme: SpotlightTheme): SelectOptions =
 };
 
 // TODO: exclude specified years
-export const createYearOptionsFromRange = (startYear = 0, endYear = 0, _exclude?: number[]): SelectOptions => {
+export const createYearOptionsFromRange = (startYear = 0, endYear = 0, excludedYear: string): SelectOptions => {
```
Comment (edwinmp) [inline utils/select.ts:32]: excludedYears... plural... and it should be optional.

File: utils/select.ts [hunk 5]
```diff
@@ -29,24 +29,30 @@ const createIndicatorOptionsFromTheme = (theme: SpotlightTheme): SelectOptions =
...
+export const createYearOptionsFromIndicator = ({
+  start_year,
+  end_year,
+  excluded_years
+}: SpotlightIndicator): SelectOptions => {
   const startYear = start_year || end_year || 0;
   const endYear = end_year || start_year || 0;
 
-  return createYearOptionsFromRange(startYear, endYear);
+  return createYearOptionsFromRange(startYear, endYear, excluded_years as string);
```
Comment (edwinmp) [inline utils/select.ts:55]: No need to cast.

File: utils/select.ts [hunk 6]
```diff
@@ -29,24 +29,34 @@ const createIndicatorOptionsFromTheme = (theme: SpotlightTheme): SelectOptions =
 };
 
 // TODO: exclude specified years
-export const createYearOptionsFromRange = (startYear = 0, endYear = 0, _exclude?: number[]): SelectOptions => {
+export const createYearOptionsFromRange = (
+  startYear = 0,
+  endYear = 0,
+  excludedYears: string | undefined
```
Comment (edwinmp) [inline utils/select.ts:35]: This is long ... optional is `excludedYears?: string`

Comment (mariamiah) [reply utils/select.ts:35]: Okay

**Pull request merged**